.github/container/test-pax.sh

```diff
@@ -136,6 +148,21 @@ while [ : ]; do
     esac
 done
 
+# Set hlo dump folder after output folder is set.
+HLO_DIR=${OUTPUT}/hlo
+export BASE_XLA_FLAGS="${BASE_XLA_FLAGS:---xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR}}"
```

Comment:
Could you please explain logic here: is `BASE_XLA_FLAGS` is set, than you always skip setting `HLO_DIR`?

Reply:
dumping the hlo is enabled by default in BASE_XLA_FLAGS, and BASE_XLA_FLAGS are appended to XLA_FLAGS env var.  if user wants to test fmha then BASE_XLA_FLAGS_FMHA is added and appended to XLA_FLAGS. The idea is to preserve the env var XLA_FLAGS before execution of this script.

Reply:
OK, let me clarify my question:
line 150 literally means:
```
if [[ "$BASE_XLA_FLAGS" == "" ]]; then
      BASE_XLA_FLAGS = "--xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR}}"
fi
```
Meaning, that if `BASE_XLA_FLAGS` is already set (by any previous scripts, or globally in the system, etc), ${HLO_DIR} will not have any effect at all.

Is that expected behaviour?

Reply:
And why do you `export` it? You use it only locally.

Reply:
The mechanism was added as per the review comment of same PR for t5x: https://github.com/NVIDIA/JAX-Toolbox/pull/442#discussion_r1468912293
refer to the discussion for details of the implementation.
The implementation BASE_XLA_FLAGS="${BASE_XLA_FLAGS:---xla_dump_hlo_as_text --xla_dump_to=${HLO_DIR}}" means update the BASE_XLA_FLAGS with previous definition if any and append xla dump hlo flags to the env vars. This also gives us the flexibility of "zero out" the env var in this script without modifying code in this script by just doing BASE_XLA_FLAGS=""